### DISCUSSION TITLE
demo-games / examples

### DISCUSSION THREAD

**Discussion by tenpaMk2** [excaliburjs/Excalibur#2281] [Show and tell]
Hi developers. Thanks for nice gaming library.
I made some demo games / examples.

- [alpha-adjust](https://tenpamk2-blog.netlify.app/apps/excalibur-alpha-adjust/)
- [asteroid](https://tenpamk2-blog.netlify.app/apps/excalibur-asteroid)
- [clocks](https://tenpamk2-blog.netlify.app/apps/excalibur-clocks)

- [sourcecode](https://github.com/tenpaMk2/excalibur-examples)

The original ideas are from [digitsensitive](https://github.com/digitsensitive/phaser3-typescript).

I'm now studying how to use `Tilemap` .

**Comment by eonarheim** [comment]
Hi @tenpaMk2 thanks for the kind words!

Also thanks for the example games these are great!

**Comment by tenpaMk2** [comment]
Hi. I'm glad to see the development of Excalibur.js very active!!
I added snake-game in my examples.

- [Play!!](https://tenpamk2-blog.netlify.app/apps/excalibur-snake/)
- [source-code](https://github.com/tenpaMk2/excalibur-examples)

I found unexpected behavior.
If you play by iPhone(iOS latest chrome or safari), "GameOver" text does not appear.
If anyone knows cause, please advise me.

**Reply by eonarheim** [reply]
@tenpaMk2 Thanks for the snake game example! This is really cool!

In the `game-over.ts` try using `halfDrawHeight` and `halfDrawHeight` which will use the logical resolution. The reason it's different on iPhone is the internal HTML canvas width/height will change depending on the the screen's DPI and on iPhone I think it's HiDPI at 2x, but on a standard definition screen it would be the center.

```typescript
      x: engine.halfDrawWidth,
      y: engine.halfDrawHeight,
```

Apologies, the current excalibur properties `drawWidth` and `canvasWidth` don't have great names and it isn't clear what they mean exactly. I'm very open to suggestions

**Reply by tenpaMk2** [reply]
@eonarheim Thanks advice!!

Oops!! I knew there are some kind of width/height such as `draw***` and `canvas***` .
But I mistook it... I guess I put too much trust in the VSCode complement. sorry.

I fixed the bug and re-upload example.
